src/components/footer/_macro.njk

```diff
@@ -13,22 +13,38 @@
         {% if params.lang == 'cy' %}
             {{-
                 onsIcon({
-                    "iconType": 'ons-logo-cy',
+                    "iconType": 'ons-logo-stacked-cy' if params.logo else 'ons-logo-cy',
                     "altText": 'Swyddfa Ystadegau Gwladol',
-                    "altTextId": 'ons-logo-cy-footer-alt'
+                    "altTextId": 'ons-logo-cy-footer-alt',
+                    "classes": 'ons-u-mr-xxs' if params.logo else ''
                 })
             -}}
         {% else %}
             {{-
                 onsIcon({
-                    "iconType": 'ons-logo-en',
+                    "iconType": 'ons-logo-stacked-en' if params.logo else 'ons-logo-en' ,
                     "altText": 'Office for National Statistics',
-                    "altTextId": 'ons-logo-en-footer-alt'
+                    "altTextId": 'ons-logo-en-footer-alt',
+                    "classes": 'ons-u-mr-xxs' if params.logo else ''
                 })
             -}}
         {% endif %}
     {% endset %}
 
+    {% set additionalLogo %}
```

Comment:
What might make this a bit clearer to read is to define this at the top of the macro inside a `if params.logo` then where you do checks for `params.logo` you can check for `additionalLogo` instead.